src/modules/rlang.rs

```diff
@@ -0,0 +1,105 @@
+use super::{Context, Module, RootModuleConfig};
+use crate::formatter::VersionFormatter;
+
+use crate::configs::rlang::RLangConfig;
+use crate::formatter::StringFormatter;
+
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("rlang");
+    let config: RLangConfig = RLangConfig::try_load(module.config);
+
+    let is_r_project = context
+        .try_begin_scan()?
+        .set_files(&config.detect_files)
+        .set_extensions(&config.detect_extensions)
+        .set_folders(&config.detect_folders)
+        .is_match();
+    if !is_r_project {
+        return None;
+    }
+
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        formatter
+            .map_meta(|var, _| match var {
+                "symbol" => Some(config.symbol),
+                _ => None,
+            })
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "version" => {
+                    let r_version = get_r_version(context)?;
+                    VersionFormatter::format_module_version(
+                        module.get_name(),
+                        &r_version,
+                        config.version_format,
+                    )
+                    .map(Ok)
+                }
+                _ => None,
+            })
+            .parse(None)
+    });
+
+    module.set_segments(match parsed {
+        Ok(segments) => segments,
+        Err(error) => {
+            log::warn!("Error in module `rlang`:\n{}", error);
+            return None;
+        }
+    });
+
+    Some(module)
+}
+
+fn get_r_version(context: &Context) -> Option<String> {
+    let r_version = context.exec_cmd("R", &["--version"])?.stderr;
+    parse_version(&r_version)
+}
+
+fn parse_version(r_version: &str) -> Option<String> {
+    r_version
+        .lines()
+        // take first line
+        .next()
```

Comment:
Use `?` here so you can simply invoke `split_whitespace` instead of chaining with `and_then`. It is more readable and in line with version extractors present in other modules.